control.go

```diff
@@ -212,12 +223,18 @@ func (tcb *ControlBlock) rcvSynRcvd(seg Segment) (pending Flags, err error) {
 
 func (tcb *ControlBlock) rcvEstablished(seg Segment) (pending Flags, err error) {
 	flags := seg.Flags
-	pending = FlagACK
-	if flags.HasAny(FlagFIN) {
-		// See Figure 5: TCP Connection State Diagram of RFC 9293.
-		tcb.state = StateCloseWait
-		tcb.pending[1] = FlagFIN // Queue FIN for after the CloseWait ACK.
+
+	dataToAck := seg.DATALEN > 0
```

Comment:
This line contains the logic to prevent duplicate acks. The rest of the PR contains tests added, and some RFC compliant changes such as challenge ack and RST handling.